src/resolve/resolved-etc-hosts.c

```diff
@@ -440,7 +440,7 @@ int manager_etc_hosts_lookup(Manager *m, DnsQuestion* q, DnsAnswer **answer) {
 
                 if (IN_SET(t->type, DNS_TYPE_A, DNS_TYPE_ANY))
                         found_a = true;
-                if (IN_SET(t->type, DNS_TYPE_AAAA, DNS_TYPE_ANY))
+                if (IN_SET(t->type, DNS_TYPE_AAAA, DNS_TYPE_ANY) && socket_ipv6_is_enabled())
```

Comment:
i am not convinced we should do this. if we synthesize localhost, then yes, we should suppress the AAAA record if ipv6 is off. But I am pretty sure we should not second guess /etc/hosts. People should not list ::1 there if they don#t want it resolved.

Quite frnakly, they should drop any entry for "localhost" altogether, and let us synthesize it.